test/mpi/comm/cmsplit_type.c

```diff
@@ -55,6 +55,38 @@ int main(int argc, char *argv[])
         MPI_Comm_free(&comm);
     }
 
+    /* Check to see if MPI_COMM_TYPE_HW_GUIDED works correctly */
+    int num_subsets = 0;
+    for (i = 0; split_topo[i]; i++) {
+        MPI_Info_create(&info);
+        MPI_Info_set(info, "mpi_hw_resource_type", split_topo[i]);
+        MPI_Comm_split_type(MPI_COMM_WORLD, MPI_COMM_TYPE_HW_GUIDED, 0, info, &comm);
+        if (comm != MPI_COMM_NULL) {
+            int newsize;
+            MPI_Comm_size(comm, &newsize);
+            if (newsize < world_size) {
+                num_subsets++;
+            }
```

Comment:
Can we report the `split_topo` name of failed subset?

Reply:
If the returned comm is not `COMM_NULL` but `newsize` >= `world_size`, do we need report it as an error?

Reply:
It is allowed to return a comm with the same size as `wolrd_size`, for example, split by "node" and every process is on the same node. Returning either `COMM_NULL` or `world_size` is not strictly an error.

So it is tricky to do a good test here. Actually, if we do not test with `MPIR_CVAR_ODD_EVEN_CLIQUES=1`, we would probably always get split of world size (since typically we do not bind processes during tests).

But the tests without any validation seem useless, thus I take this approach that assumes some splits should end up with strictly sub-splits.

Reply:
I still don't get the idea why we check `num_subsets == 0` and post only a "warning" message for it (not error).

Maybe we could pass `mpiexec` argument to set the binding, and check the hint properly?

Reply:
> I still don't get the idea why we check `num_subsets == 0` and post only a "warning" message for it (not error).

Oh, you mean I forgot to increment `errs`? I should do that.

> Maybe we could pass mpiexec argument to set the binding, and check the hint properly?

We'll need to fix the hardware platforms where tests are run to effectively test, and the functionality will depend on the `hwloc` support which only works well on Linux I believe.

Reply:
Alright, I rewrote the tests (for `MPI_COMM_TYPE_HW_GUIDED`)  following the same tests as `shmem_topo` below.

Reply:
I think it might be better to report error "Expected comm size to be at most the world size" (not node size) in case we extend the implementation to support network topo.